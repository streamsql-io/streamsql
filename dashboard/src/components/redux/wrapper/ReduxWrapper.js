import React from "react";
import { Provider } from "react-redux";

const ReduxWrapper = ({ children, store }) => (
  <Provider store={store}>{children}</Provider>
);

export default ReduxWrapper;
