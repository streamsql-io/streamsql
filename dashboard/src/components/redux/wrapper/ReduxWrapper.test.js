import React from "react";
import { configure, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { connect } from "react-redux";
import { configureStore, createSlice } from "@reduxjs/toolkit";

import ReduxWrapper from "./ReduxWrapper.js";

configure({ adapter: new Adapter() });

describe("ReduxWrapper", () => {
  const exampleSlice = createSlice({
    name: "example",
    initialState: {
      value: 5,
    },
  });

  const testStore = configureStore({
    reducer: exampleSlice.reducer,
  });

  const StatefulDiv = ({ valueProp }) => <div>{valueProp}</div>;

  const mapStateToProps = (state) => ({
    valueProp: state.value,
  });

  const ConnectedDiv = connect(mapStateToProps)(StatefulDiv);

  it("Add store to context to allow connect to work", () => {
    expect(
      mount(
        <ReduxWrapper store={testStore}>
          <ConnectedDiv />
        </ReduxWrapper>
      )
    ).toMatchSnapshot();
  });
});
