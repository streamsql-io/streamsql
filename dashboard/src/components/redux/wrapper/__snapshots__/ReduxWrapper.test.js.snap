// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReduxWrapper Add store to context to allow connect to work 1`] = `
<ReduxWrapper
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <Connect(StatefulDiv)>
      <StatefulDiv
        dispatch={[Function]}
        valueProp={5}
      >
        <div>
          5
        </div>
      </StatefulDiv>
    </Connect(StatefulDiv)>
  </Provider>
</ReduxWrapper>
`;
